crates/transaction-generator-lib/src/call_custom_modules.rs

```diff
@@ -83,18 +83,44 @@ impl CallCustomModulesCreator {
     ) -> Self {
         let mut rng = StdRng::from_entropy();
         assert!(accounts.len() >= num_modules);
-        let mut requests = Vec::with_capacity(accounts.len());
+        let mut publish_requests = Vec::with_capacity(accounts.len());
         let mut package_handler = PackageHandler::new();
         let mut packages = Vec::new();
         for account in accounts.iter_mut().take(num_modules) {
             let package = package_handler.pick_package(&mut rng, account);
             let txn = package.publish_transaction(account, &init_txn_factory);
-            requests.push(txn);
+            publish_requests.push(txn);
             packages.push((package, account.address()));
         }
-        info!("Publishing {} packages", requests.len());
-        txn_executor.execute_transactions(&requests).await.unwrap();
-        info!("Done publishing {} packages", requests.len());
+        info!("Publishing {} packages", publish_requests.len());
+        txn_executor
+            .execute_transactions(&publish_requests)
+            .await
+            .unwrap();
+        info!("Done publishing {} packages", publish_requests.len());
+
+        // For Token V1/V2 transactions, we first need to initialize collections before generating mint/transfer transactions.
+        // The initial_entry_point is the initialize_collection method for the Token transactions.
+        let mut initial_requests = Vec::with_capacity(accounts.len());
+        if let Some(initial_entry_point) = entry_point.initialize_entry_point() {
+            for account in accounts.iter_mut().take(num_modules) {
+                let package = package_handler.pick_package(&mut rng, account);
+                let request = package.use_specific_transaction(
+                    initial_entry_point,
+                    account,
+                    &init_txn_factory,
+                    Some(&mut rng),
+                    Some(&account.address()),
+                );
+                initial_requests.push(request);
+            }
+            info!("Initializing {} collections", initial_requests.len());
```

Comment:
My understanding of NFT mint is that there is one collection that is used to mint tokens, here we are initializing multiple collections. Is this how the Reinfield setup looked like? cc - @igor-aptos ?

Reply:
Here the code is initializing a collection for each package published in the previous for loop. In our case, we are only publishing Simple.move package. So, this code will always initialize only one collection.

Reply:
@sitalkedia Addressing your previous comment. Some forge tests were done for non-fungible token mint transaction, and some tests were done for fungible token mint transaction. For each transaction type, different forge tests yield roughly the same results.

Reply:
@vusirikala - Can you kick off a clean run for non-fungible token serial mint transaction?

Reply:
@sitalkedia Started the forge test again for NFT mints.